### PR TITLE
Clean up docs workflow

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description

~This is a bit of a tricky one~ Leaving it for now, just a really simple PR now

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: .github/workflows/check-links.yaml [hunk 1]
```diff
@@ -39,15 +29,12 @@ jobs:
...
         run: make install-site-deps
-      - name: Build docs for link check
+      - name: Build docs
         run: make build-pages
-        # Using liche action to check generated HTML site
-      - name: Link Checker (generated site)
-        id: lc
-        uses: peter-evans/link-checker@v1
+      - name: Check links
+        uses: gaurav-nelson/github-action-markdown-link-check@v1
```
Comment (michaelbeaumont) [inline .github/workflows/check-links.yaml:37]: Can anyone tell me why we were using two link checkers?

File: .github/workflows/markdown-link-check-config.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
     "ignorePatterns": [
         { "pattern": "^https://github.com/\\S+/\\S+/(issues|pull)/[0-9]+" },
+        { "pattern": "^https://twitter.com" },
         { "pattern": "^mailto:" },
-        { "pattern": "^https://circleci.com/workflow-run/02d8b5fb-bc7f-404c-9051-68307c124649" },
```
Comment (michaelbeaumont) [inline .github/workflows/markdown-link-check-config.json:5]: I don't think we need this anymore...

File: .github/workflows/check-links.yaml (hunk 1 above)
Comment (Callisto13) [reply .github/workflows/check-links.yaml:37]: does seem weird, they both came in with the same commit so it looks deliberate. do they each offer different features?

Comment (michaelbeaumont) [reply .github/workflows/check-links.yaml:37]: yeah, I think it has to do with checking `userdocs` as HTML and other `.md` files as Markdown...

**Comment by michaelbeaumont** [comment]
Massively decreased the scope here, the one action isn't maintained anymore but the replacement has some issue so leaving it for now.

File: .github/workflows/check-links.yaml (hunk 1 above)
Comment (michaelbeaumont) [reply .github/workflows/check-links.yaml]: leaving it for now

**Review by aclevername — APPROVED** [review]
LGTM

**Pull request merged**